ctk/common/src/main/kotlin/org/codice/compliance/utils/TestCommon.kt

```diff
@@ -228,3 +228,5 @@ class LazyVar<T>(val init: () -> T) : ReadWriteProperty<Any?, T> {
         this.value = Optional.of(value)
     }
 }
+
+data class NodeDecorator(val node: Node, var hasEncryptedAssertion: Boolean = false)
```

Comment:
✏️ This isn't really a decorator, as it doesn't act like a `Node` itself.

I think you could reduce the scope of your changes by 
1. Making this a true decorator by having this class extend `Node`.
2. Adding an extension function on `Node` to answer the question of `hasEncryptedAssertion`
    a. If `this` is `NodeDecorator`, return the value of `hasEncryptedAssertion`
    b. Else, return false or throw a not-implemented exception or log a message and return false, whatever makes the most sense
3. Then you just keep using `Node`s everywhere in param and return value types and only have to be explicit on the lines where you construct the decorators

Reply:
The main concern I have about "extending `Node`" is that it's actually an interface that defines 37 methods, so we'd have to implement all 37 of those methods, delegating to the decorated `Node` in each one.

Reply:
I'm going with a `NodeWrapper` instead to avoid that ☝️ We can talk about naming after I get the commit up, if you guys don't like it 😄